workos/session.py

```diff
@@ -0,0 +1,196 @@
+                RefreshTokenAuthenticationResponse,
+                self.user_management.authenticate_with_refresh_token(
+                    refresh_token=session["refresh_token"],
+                    organization_id=organization_id,
+                    session={"seal_session": True, "cookie_password": cookie_password},
+                ),
+            )
+
+            self.session_data = str(auth_response.sealed_session)
+            self.cookie_password = (
+                cookie_password if cookie_password is not None else self.cookie_password
+            )
+
+            signing_key = self.jwks.get_signing_key_from_jwt(auth_response.access_token)
+
+            decoded = jwt.decode(
+                auth_response.access_token,
+                signing_key.key,
+                algorithms=self.jwk_algorithms,
+            )
+
+            return RefreshWithSessionCookieSuccessResponse(
+                authenticated=True,
+                sealed_session=str(auth_response.sealed_session),
+                session_id=decoded["sid"],
+                organization_id=decoded.get("org_id", None),
+                role=decoded.get("role", None),
+                permissions=decoded.get("permissions", None),
+                entitlements=decoded.get("entitlements", None),
+                user=auth_response.user,
+                impersonator=auth_response.impersonator,
+            )
+        except Exception as e:
+            return RefreshWithSessionCookieErrorResponse(
+                authenticated=False, reason=str(e)
+            )
+
+    def get_logout_url(self) -> str:
+        auth_response = self.authenticate()
+
+        if isinstance(auth_response, AuthenticateWithSessionCookieErrorResponse):
+            raise ValueError(
+                f"Failed to extract session ID for logout URL: {auth_response.reason}"
+            )
+
+        result = self.user_management.get_logout_url(
+            session_id=auth_response.session_id
+        )
+        return str(result)
+
+    def is_valid_jwt(self, token: str) -> bool:
+        try:
+            signing_key = self.jwks.get_signing_key_from_jwt(token)
+            jwt.decode(token, signing_key.key, algorithms=self.jwk_algorithms)
+            return True
+        except jwt.exceptions.InvalidTokenError:
+            return False
+
+    @staticmethod
+    def seal_data(data: Dict[str, Any], key: str) -> str:
+        fernet = Fernet(key)
+        # Encrypt and convert bytes to string
+        encrypted_bytes = fernet.encrypt(json.dumps(data).encode())
+        return encrypted_bytes.decode("utf-8")
+
+    @staticmethod
+    def unseal_data(sealed_data: str, key: str) -> Dict[str, Any]:
+        fernet = Fernet(key)
+        # Convert string back to bytes before decryption
+        encrypted_bytes = sealed_data.encode("utf-8")
+        decrypted_str = fernet.decrypt(encrypted_bytes).decode()
+        return cast(Dict[str, Any], json.loads(decrypted_str))
```

Comment:
If these are all private, we may want to mark them with a `_` prefix.

Reply:
They're not really private since they can still be accessed statically. Should we still add the prefix?

Reply:
Good question, up to you. "Private" is really just a suggestion anyway. It has no real encapsulation benefits in Python. `is_valid_jwt` could at least use it though. More-so just helps us so we can consider these methods internal if we need to update.